tools/preflight/helper.go

```diff
@@ -201,6 +219,87 @@ func getVersionsOfGroup(grp string) ([]string, error) {
 	return apiVerList, nil
 }
 
+//  checkAndCreateVolumeSnapshotCRDs checks and creates volumesnapshot and related CRDs if not present on cluster.
+func checkAndCreateVolumeSnapshotCRDs(ctx context.Context) error {
+	var err error
+
+	serverVersion, sErr := discClient.ServerVersion()
+	if sErr != nil {
+		return sErr
+	}
+
+	crdObj, prefCRDVersion, gErr := getPrefVersionCRDObj(serverVersion.String())
+	if gErr != nil {
+		return gErr
+	}
+
+	for _, crd := range volumesnopshotcrds {
+		if err = runtimeClient.Get(ctx, client.ObjectKey{Name: crd}, crdObj); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				return fmt.Errorf("error getting volume snapshot class CRD :: %s", err.Error())
+			}
+			logrus.Infof("volume snapshot CRD: %s not found on cluster. Attempting installation...", crd)
```

Comment:
Does this log statement log to both stdout and the log file of preflight?

A logger is associated with a preflight `Run` struct. Pass that logger to this func. So that, the logs will also get written to the log file.